modules/ns/natives/player_memory.cpp

```diff
@@ -378,6 +388,9 @@ static cell AMX_NATIVE_CALL ns_remove_upgrade(AMX *amx, cell *params)
 
 	if (bfound)
 	{
+		void *pTechTree = reinterpret_cast<char*>(player->GetEdict()->pvPrivateData) + MAKE_OFFSET(COMBAT_TECHTREE);
```

Comment:
Note: This is to access player's private data `AvHTechTree  mCombatNodes;` Then, we can do afterward `mCombatNodes.SetResearchDone()`